### DISCUSSION TITLE
Ideas and new features

### DISCUSSION THREAD

**Discussion by showyaseen** [showyaseen/intelli-builder#4] [Ideas]
Let’s discuss the next features and improvements for the plugin. Feel free to share any ideas or suggestions you have!

**Comment by elettrona** [comment]
Browser detection: the plug-in seems not to properly work when changing browser language.
Tried with Chrome and Edge without success. It seems to work with Brave Browser - found on brave.com - instead.

Tried both by adding a new language to browser's language preferences, and adding just the desired language removing all the others.

Another idea shoule be "apply condition to nested blocks"
if you have a group, quote, details, queryloop ... the nested blocks inherit the visibility conditions (should be a toggleable flag). But before going ahead I'd like to be sure browser detection works.
This should be a must-have for Full Site Editing I think.

**Reply by elettrona** [reply]
I'd also suggest another condition:
check WordPress interface's language - php function should be "get_current_locale".
So, the condition should be: "if get current locale is set to [language] then show -or hide- block.
Then, operators such as "and, or, is, is not". To have a set of rules for blocks.
And, in the actions, even change wordpress language (switch_to_locale function) should be implemented too.
...And, don't forget this...
"is (or isn't" in the RSS feed. I mean: show or hide content from RSS in order to encourage readers to subscribe to feeds.

**Reply by showyaseen** [reply]
Hey @talksina, thanks for the feedback!

I'll do some more testing with Chrome and Edge to figure out what's going on with the browser detection.

I also like your idea about adding a toggle for nested blocks to inherit visibility conditions. That would be super helpful, especially for FSE. I’ll look into it and see if we can get it into the next release.

Appreciate the suggestions and keep them coming!

**Comment by elettrona** [comment]
@showyaseen another interesting condition for blocks would also be "show to RSS readers only".
- WordPress condition is "is_feed" or "!is_feed" with exclamation point if you want to instruct your code negatively ("if NOT is_feed").

There are lots of use cases for this one as you can give some extra-contents to people reading through RSS-based systems (I still have to verify if it works with Fediverse).

I'm in the experimenting mood.

**Comment by elettrona** [comment]
[language-shortcode-with-locale.txt](https://github.com/user-attachments/files/17006969/language-shortcode-with-locale.txt)
Hello, through both AI and Google searches I have managed to play around and create a couple shortcodes through which I control content via browser languages. It works. I wanted to share it hoping it could help you for language browser detection. Hth. My 2 cents.
